pkg/kn/commands/types.go

```diff
@@ -47,3 +57,33 @@ func GetConfig() (serving.ServingV1alpha1Interface, error) {
 	}
 	return client, nil
 }
+
+// GetCurrentNamespace returns current namespace from kubeconfig.
+func (c *KnParams) GetCurrentNamespace() (string, error) {
+	// Make sure that KubeCfgFile is initialized.
+	if c.KubeCfgFile == "" {
+		if err := c.initKubeConfig(); err != nil {
+			return "", err
+		}
+	}
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+	rules.ExplicitPath = c.KubeCfgFile
```

Comment:
see https://github.com/k14s/kapp/blob/0c91e8a2208ff9a8f9d92f8c01d20dc9dd21256b/pkg/kapp/cmd/core/config_factory.go#L60 for example. we dont need to specify ExplicitPath unless it's given as an explicit option. also KUBECONFIG env var will be picked up by default.